--- tccgen.c	2022-06-20 10:51:37.000000000 +0530
+++ ./tcc-0.9.27/tccgen.c	2017-12-17 13:57:05.000000000 +0530
@@ -17,7 +17,7 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
-#include <assert.h>
+
 #include "tcc.h"
 
 /********************************************************/
@@ -5115,13 +5115,8 @@
         } else if (tok == '.' || tok == TOK_ARROW || tok == TOK_CDOUBLE) {
             int qualifiers;
             /* field */ 
-            if (tok == TOK_ARROW) {
+            if (tok == TOK_ARROW) 
                 indir();
-            }else if(tok == '.'){
-                if ((vtop->type.t & VT_BTYPE) == VT_PTR) {
-+                    indir();
-+                }
-            }
             qualifiers = vtop->type.t & (VT_CONSTANT | VT_VOLATILE);
             test_lvalue();
             gaddrof();
